File: Whatsapp/OpenWA/sdk/php/src/Resources/ProfileResource.php
<?php

declare(strict_types=1);

namespace OpenWA\Resources;

use OpenWA\Http\HttpExecutor;

/**
 * Profile resource — the connected account's own profile.
 *
 * Backed by src/modules/profile/profile.controller.ts.
 */
class ProfileResource
{
    private HttpExecutor $http;

    public function __construct(HttpExecutor $http)
    {
        $this->http = $http;
    }

    /**
     * Set the account display name. Requires an OPERATOR-level key.
     *
     * @return array<string,mixed>
     */
    public function setProfileName(string $sessionId, string $name): array
    {
        return $this->http->request('PUT', "/api/sessions/{$this->http->encodeSegment($sessionId)}/profile/name", [], ['name' => $name]);
    }

    /**
     * Set the account about/status text; an empty string clears it. Requires an
     * OPERATOR-level key.
     *
     * @return array<string,mixed>
     */
    public function setProfileStatus(string $sessionId, string $status): array
    {
        return $this->http->request('PUT', "/api/sessions/{$this->http->encodeSegment($sessionId)}/profile/status", [], ['status' => $status]);
    }

    /**
     * Set the account profile picture. Body is either {url} or
     * {base64, mimetype}. Requires an OPERATOR-level key.
     *
     * @param array<string,mixed> $body
     * @return array<string,mixed>
     */
    public function setProfilePicture(string $sessionId, array $body): array
    {
        return $this->http->request('PUT', "/api/sessions/{$this->http->encodeSegment($sessionId)}/profile/picture", [], $body);
    }
}

File: OpenWA/sdk/java/src/main/java/com/rmyndharis/openwa/resources/ProfileResource.java
package com.rmyndharis.openwa.resources;

import static com.rmyndharis.openwa.http.Http.encodeSegment;

import com.rmyndharis.openwa.OpenWAClient;
import com.rmyndharis.openwa.http.HttpMethod;
import com.rmyndharis.openwa.model.SetProfileNameRequest;
import com.rmyndharis.openwa.model.SetProfilePictureRequest;
import com.rmyndharis.openwa.model.SetProfileStatusRequest;
import com.rmyndharis.openwa.model.SuccessResult;

/** Profile resource — the linked account's display name, status text and picture. */
public final class ProfileResource {
    private final OpenWAClient client;

    public ProfileResource(OpenWAClient client) {
        this.client = client;
    }

    /** Set the account display name. */
    public SuccessResult setProfileName(String sessionId, String name) {
        return client.request(
            HttpMethod.PUT,
            "/api/sessions/" + encodeSegment(sessionId) + "/profile/name",
            null,
            new SetProfileNameRequest(name),
            SuccessResult.class);
    }

    /** Set the account about/status text (an empty string clears it). */
    public SuccessResult setProfileStatus(String sessionId, String status) {
        return client.request(
            HttpMethod.PUT,
            "/api/sessions/" + encodeSegment(sessionId) + "/profile/status",
            null,
            new SetProfileStatusRequest(status),
            SuccessResult.class);
    }

    /** Set the account profile picture from a {@code url} or a {@code base64} + {@code mimetype} pair. */
    public SuccessResult setProfilePicture(String sessionId, SetProfilePictureRequest body) {
        return client.request(
            HttpMethod.PUT,
            "/api/sessions/" + encodeSegment(sessionId) + "/profile/picture",
            null,
            body,
            SuccessResult.class);
    }
}
